dFusion/dFusionSpec.md

```diff
@@ -79,11 +79,92 @@ To summarize, here is a list of state that is stored inside the smart contract:
 ## Batch Auction Workflow
 
 The trading workflow consists of the following sequential processes:
+
+0. Account opening, deposits & withdrawals
 1. On-Chain order collection
 2. Finding the batch price: optimization of batch trading surplus (off-chain)
 3. Verifying batch price and trade execution (zkSnark)
 4. Restart with step 1
 
+### Account opening & token registration
+
+
+#### Registering accounts
+The contract specifies a constant number of accounts, `K`, that can be opened by anyone by providing an `accountIndex` such that `1 <= accountIndex <= K`. 
+The account index is referred to, synonomously, as the account ID.
+
+Upon opening an account, the contract verifies that
+    - there is no account currently occupying the requested index and
+    - the sender does not currently occupy an account slot.
+
+On the level of contract storage, these contraints imply a bijective mapping {1, 2, ..., K} <-> addresses.
+
+Note that: Registering accounts by specified index (rather than incrementing) enables the possiblity for accounts to be closed and account slots to be made available.
+
+TODO - holding an account will likely incur some kind of fee.
```

Comment:
If this works for you: We should start with the assumption that account lease will be a fixed flat fee per batch in a single denominated currency, eg ETH, for which the account holder can place X orders in a batch. X = slots/accounts.

This isn't all that likely to work out, but its the simplest thing there is.

Reply:
Imagine the following scenario:

We do not charge rent for the accounts. Somebody creates `MAX_ACCOUNTS` different Ethereum address, fills all the account slots and  never participates in a single auction.

Reply:
Right, that's why I would charge a lease for the accounts and allocate slots by account. Not at all efficient, but there are multiple ways to expand from there. The account/slot hogging scenario needs to be addressed, but right now for the BVP I would like to think through "what's the simplest fee structure to keep this running, assuming away all attacks". Not an easy thing after thinking about attack vectors for a few months. 

Defenses against the kinds of attacks we're envisioning can come later (in MarchVP). This is all a tradeoff between protection and usability.

Reply:
So is there a suggested change here or will this merged as is?